### ISSUE TITLE
Create Governance Dictionary

### ISSUE THREAD

**Issue by chapati23** [mento-protocol/mento-core#326]
### Description
Create a Notion Page that serves as a ELI5 type of dictionary to explain roles and responsibilities of each component in our governance system.

### Acceptance Criteria
- [x] A new **Governance Dictionary** Notion Page under Engineering has been created (=> https://www.notion.so/Governance-Dictionary-1bd33e891eb0401f8d4f455dd773648f?pvs=4)

**Comment by bayological** [comment]
Nice work and initiative🔥. This doc will be especially useful when it's time to update our public-facing docs.
